src/mesh/simplex_refiner.C

```diff
@@ -0,0 +1,598 @@
+// The libMesh Finite Element Library.
+// Copyright (C) 2002-2024 Benjamin S. Kirk, John W. Peterson, Roy H. Stogner
+
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+
+// Local includes
+#include "libmesh/simplex_refiner.h"
+
+#include "libmesh/boundary_info.h"
+#include "libmesh/face_tri3.h"
+#include "libmesh/int_range.h"
+#include "libmesh/libmesh_logging.h"
+#include "libmesh/mesh_base.h"
+#include "libmesh/mesh_communication.h"
+#include "libmesh/mesh_tools.h"
+#include "libmesh/node.h"
+#include "libmesh/parallel_algebra.h"
+#include "libmesh/remote_elem.h"
+#include "libmesh/unstructured_mesh.h"
+#include "libmesh/utility.h"
+
+// TIMPI includes
+#include "timpi/parallel_implementation.h"
+#include "timpi/parallel_sync.h"
+
+
+namespace libMesh
+{
+
+//-----------------------------------------------------------------
+// Mesh refinement methods
+SimplexRefiner::SimplexRefiner (UnstructuredMesh & mesh) :
+  _mesh(mesh),
+  _desired_volume(0.1)
+{}
+
+
+
+bool SimplexRefiner::refine_elements()
+{
+  if (_mesh.is_prepared())
+    _mesh.prepare_for_use();
```

Comment:
Are you missing a `!` here?

Reply:
I assume `!` is a euphemism for "brain cell", in which case the answer is yes.

If you just meant a literal `!` the answer is also yes.

Reply:
@permcody Why are you always preparing?